qgrid/grid.py

```diff
@@ -21,9 +23,125 @@ def template_contents(filename):
+
+        Parameters
+        ----------
+        optname : str
+            The name of the option to override
+        optvalue : object
+            The new value
+        """
+        self._js_options[optname] = optvalue
+
+    def set_defaults(self, remote_js=None, precision=None, js_options=None):
+        """
+        Set a default value to be passed to Python SlickGrid instances.
+
+        See documentation for `show_grid` for more info on configurable values.
+        """
+        if remote_js is not None:
+            self._remote_js = remote_js
+        if precision is not None:
+            self._precision = precision
+        if js_options is not None:
+            self._js_options = js_options
+
+    @property
+    def js_options(self):
+        return self._js_options
+
+    @property
+    def remote_js(self):
+        return self._remote_js
+
+    @property
+    def precision(self):
+        return self._precision or pd.get_option('display.precision') - 1
+
+defaults = _DefaultSettings()
+set_defaults = defaults.set_defaults
+set_js_option = defaults.set_js_option
+
+
+def show_grid(data_frame, remote_js=None, precision=None, js_options=None):
+    """
+    Main entry point for rendering DataFrames as SlickGrids.
+
+    Parameters
+    ----------
+    remote_js : bool
+        Whether to load slickgrid.js from a local filesystem or from a
+        remote CDN.  Loading from the local filesystem means that SlickGrid
+        will function even when not connected to the internet, but grid
+        cells created with local filesystem loading will not render
+        correctly on external sharing services like NBViewer.
+    precision : integer
+        The number of digits of precision to display for floating-point
+        values.  If unset, we use the value of
+        `pandas.get_option('display.precision')`.
+    js_options : dict
+        Options to use when creating javascript SlickGrid instances.  See
+        the SlickGrid documentation for information on the available
+        options.  Default options are as follows:
+
+        {
+            'enableCellNavigation': True,
+            'fullWidthRows': True,
+            'syncColumnCellResize': True,
+            'forceFitColumns': True,
+            'rowHeight': 28,
+            'enableColumnReorder': False,
+            'enableTextSelectionOnCells': True,
+        }
+
+    See Also
+    --------
+    qgrid.set_defaults : Permanently set global defaults for `show_grid`.
+    qgrid.set_js_option : Permanently set individual Javascript options.
+    """
+
+    if remote_js is None:
+        remote_js = defaults.remote_js
+    if precision is None:
+        precision = defaults.precision
+        if not isinstance(precision, Integral):
+            raise TypeError("precision must be int, not %s" % type(precision))
+    if js_options is None:
+        js_options = defaults.js_options
+        if not isinstance(js_options, DictType):
```

Comment:
I'm tried this with Python 3 and I got an ImportError for `DictType`.  Apparently in Python 3 dict is supposed to be used instead.  It seems like that will work in Python 2 as well (I tried it out on my machine).